src/chat.rs

```diff
@@ -709,47 +709,51 @@ impl ChatId {
 
     /// Deletes a chat.
     pub async fn delete(self, context: &Context) -> Result<()> {
+        self.delete_ex(context, Sync).await
+    }
+
+    pub(crate) async fn delete_ex(self, context: &Context, sync: sync::Sync) -> Result<()> {
         ensure!(
             !self.is_special(),
             "bad chat_id, can not be a special chat: {}",
             self
         );
 
         let chat = Chat::load_from_db(context, self).await?;
-        context
-            .sql
-            .execute(
+        let sync_id = match sync {
+            Nosync => None,
+            Sync => chat.get_sync_id(context).await?,
+        };
+
+        let trans_fn = |t: &mut rusqlite::Transaction| {
+            t.execute(
                 "DELETE FROM msgs_mdns WHERE msg_id IN (SELECT id FROM msgs WHERE chat_id=?);",
                 (self,),
-            )
-            .await?;
-
-        context
-            .sql
-            .execute("DELETE FROM msgs WHERE chat_id=?;", (self,))
-            .await?;
-
-        context
-            .sql
-            .execute("DELETE FROM chats_contacts WHERE chat_id=?;", (self,))
-            .await?;
-
-        context
-            .sql
-            .execute("DELETE FROM chats WHERE id=?;", (self,))
-            .await?;
-
+            )?;
+            t.execute("DELETE FROM msgs WHERE chat_id=?;", (self,))?;
+            t.execute("DELETE FROM chats_contacts WHERE chat_id=?;", (self,))?;
+            t.execute("DELETE FROM chats WHERE id=?;", (self,))?;
+            Ok(())
+        };
+        context.sql.transaction(trans_fn).await?;
         context.emit_msgs_changed_without_ids();
 
-        context.set_config(Config::LastHousekeeping, None).await?;
-        context.scheduler.interrupt_inbox().await;
+        if let Some(id) = sync_id {
+            self::sync(context, id, SyncAction::Delete)
+                .await
+                .log_err(context)
+                .ok();
+        }
 
         if chat.is_self_talk() {
```

Comment:
I'd move this even before the actual deletion so that this message can't be missing if we fail half-way. But then the stock string should be changed to `You are deleting the \"Saved messages\" chat...` or smth like this